### PR TITLE
Oak ridge transform

### PR DESCRIPTION
**Pull request by mlhale7**
**GitHub Issue**: [249](https://github.com/DigitalLibraryofTennessee/DLTN_XSLT/issues/249)

* Other Relevant Links (OAI Feeds, DPLA Records, Metadata Mappings, DLTN Documentation, Etc.)
[http://cdm16107.contentdm.oclc.org/oai/oai.php?verb=ListSets](http://cdm16107.contentdm.oclc.org/oai/oai.php?verb=ListSets) 

## What does this Pull Request do?

This PR adds a transform for Oak Ridge Public Library as well as metadata from the two sets they have made available.

## How should this be tested?

Use the transform on one of the sets of metadata and see if DPLA's requirements are met (title, rights, link to object, link to thumbnail, provider). Quickly check to see if object/thumbnail links are correctly formatted and function. Note that Oak Ridge has not turned on IIIF, so these links are not given in the metadata.

## Additional Notes:

Conversation with Julie Forkner and Michael Stallo at Oak Ridge will take place to confirm some of the metadata decisions (for instance, if they are fine with "In Copyright"). We will also be meeting to see if they can split out their dates better in the future (one kind of date for the creation date associated with the object and another kind of date to deal with digitization / donation). For the December ingest we likely won't see any changes in the data.

## Interested parties

@markpbaggett

### CODE REVIEW

**Review by markpbaggett — COMMENTED** [review]
@mlhale7, this looks pretty good.  I'm curious about the thoughts behind positional matching on the three elements.

Also, what's the thinking here for final testing before share?  Are you thinking:

1. Make sure the transform looks okay
2. Approve and merge transform
3. Add sets to repox with new transform
4. Run check and harvest vs. MODS feed

We definitely need to run tests versus the MODS feed, and as far as I can tell, the old repox vagrant no longer works.  

If we want to test this versus real data prior to merge and approve, I could go ahead and add the sets to repox and run check and harvest versus MODS with this transform.

If not, we can approve and merge, then do that, but we definitely need to do that before we share with DPLA.  Let me know what you were thinking.

File: XSLT/oakridgedctomods.xsl [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+            <xsl:apply-templates select="dc:identifier"/>
+            
+            <!-- description containing abstract info -->
+            <xsl:apply-templates select="dc:description[1]"/>
+            
+            <!-- contributor -->
+            <xsl:apply-templates select="dc:contributor[1]"/>
+            
+            <!-- originInfo> -->
+            <xsl:apply-templates select="dc:date[1]"/>
```
[the remark this answers is not in the document]
Comment (mlhale7) [reply XSLT/oakridgedctomods.xsl:43]: For p15388cdi, there's both a digitization date (I assume) and a dateCreated date, as I alluded to in the PR description. I don't think we want to share the digitization date, but this is the only date associated with p16107coll3. I thought about restricting to values that begin '19', but then we'd have no dates for p16107coll3. I'm going to ask today if they'd like the digitization date included for p16107coll3.

For description, there's all kinds of information in this field:
`<dc:description>A sign illustrating a human figure with a globe as a head holding an atomic energy symbol and another figure dressed as a centurion with the words, &quot;for world peace . . . protect the know how.&quot;</dc:description>
<dc:description>3001 KB</dc:description>
<dc:format>JPEG image</dc:format>
<dc:description>3001 KB</dc:description>
<dc:description>Oak Ridge Public Library</dc:description>`
Oak Ridge does consistently place the most descriptive abstract first for both sets. I could work on more complex logic to exclude those that begin with numbers, etc., but I think this list would need to be added to on a case by case basis as we add sets.

For contributor, Oak Ridge has accidentally included the same value twice in p16107coll3. I don't want James A. Dougherty listed as a contributor twice for no reason.

**Comment by mlhale7** [comment]
In terms of testing, I was thinking here we needed to ensure that the sample data (actually the full set of metadata for each set) works appropriately. I assumed if the transform worked with the given data that this should indicate that it will work on Repox. I saw an approve and merge as a trigger to add the transform to the dpla server. We do need to ensure that this also will work on Repox, but I initially just envisioned this as an additional ticket.

File: XSLT/oakridgedctomods.xsl (hunk 1 above)
[the remark this answers is not in the document]
Comment (mlhale7) [reply XSLT/oakridgedctomods.xsl:43]: I just noticed that there is a second contributor in p15388cdi that the positional logic cuts off. . . I'll need to update this to look for duplicates for something else instead.

**Review by markpbaggett — APPROVED** [review]
:+1:

**Pull request merged**